File: problem003/answer003.pl
#!/usr/bin/perl

use strict;
use warnings;
use Math::Prime::Util qw(is_prime factor);
use Data::Dumper;

print "prime factors of 13195: " . Dumper(prime_factors(13195));
print "ANSWER:" . biggest_prime_factor(600851475143). "\n";

sub prime_factors {
	my ($n) = @_;

	my @factors = factor($n);
	my @this_prime_factors = map { $_ if is_prime($_) } @factors;

	return \@this_prime_factors;
}

sub biggest_prime_factor {
	my ($n) = @_;
	my $prime_factors = prime_factors($n);
	return $prime_factors->[-1];
}

File: problem003/answer003.py
#!/usr/bin/python
"""chicks' answer to Euler Project problem #3"""

def factor(n):
	factors = []
	x = 2
	while x < n:
		if n % x == 0:
			factors.append(x)
		if (x % 10000000) == 0:
			print "\tchecking " + str(x) + " for " + str(n) + " (" + str(float(x/n)) + "%)"
		x = x + 1

	print str(n) + " has " + str(len(factors)) + " factors"

	return factors

def prime_factor(n):
	any_factors = factor(n)
	prime_factors = []
	for x in any_factors:
		x_factors = factor(x)
		if len(x_factors) == 0:
			prime_factors.append(x)

	return prime_factors

print "prime factors of 13195 are " + str(prime_factor(13195))
print "prime factors of 131950 are " + str(prime_factor(131950))
print "prime factors of 600851475143 are " + str(prime_factor(600851475143))
